### ISSUE TITLE
[QUESTION]GUI Implementation for shairport

### ISSUE THREAD

**Issue by janlei0524** [mikebrady/shairport-sync#398]
Hi mike! I am really impressed with this implementation. I would like to ask some questions :)

I am planning to add a GUI for this implementation of shairport, is it possible to:
1) add a play,pause,stop,next,previous, volume up, volume down function for it?
2) to display the cover art, title,album,artist

In short, it's like a media player for shairport.

**Comment by mikebrady** [comment]
Thanks for the post.

Unfortunately, at the present time, I think this would be an almost impossible task. Shairport Sync runs as demon – it runs in a special user account and has no user interface. Therefore, the only way to put a GUI on it would be by exchanging messages with it. Unfortunately, it does not implement any form of IPC. So, it would be necessary for someone to add an IPC interface to it, e.g. dbus, and that would be a very tricky thing to do.

A further issue is that its operation is pretty closely controlled by the audio source – iTunes or the iOS device, so that the amount of control you could exercise locally would be limited.

An alternative it to develop a GUI for a system to control the source – that is, to control iTunes or whatever the source happens to be. I know that some people have been taking that approach.

Sorry if I can't be a bit more positive at this time...

**Comment by mistepien** [comment]
gui is not coherent with KISS. that's all.

**Comment by kingosticks** [comment]
I've a similar request for control but more basic. I'd like a way to tell shairport-sync to forcefully disconnect the current source, and release the playback device (perhaps additionally sending the source a busy signal?). 

In my use case I have both shairport and mopidy/mpd potentially playing audio. We currently make use of the great `run_this_before_play_begins` feature to run `mpc  stop` and ensure only one thing is playing at a time. It'd be great if this mechanism could work both ways. 

I appreciate you don't want to mess around implementing a full frontend but perhaps something simple coild be achieved with just Unix signals? Otherwise I'm thinking I could restart the daemon to achieve this but thats a huge hack. Any thoughts?

**Comment by kingosticks** [comment]
I just realised that https://github.com/mikebrady/shairport-sync/issues/389 would have been a better place to post my question, sorry.

**Comment by mikebrady** [comment]
Thanks for the comments. I'm looking into is in the context of Shairport Sync 3.

**Comment by janlei0524** [comment]
Hi mike!

Right now, I've managed to send DACP commands to the Iphone using the retrieve Active remote and DACP ID, but I used shairplay by juhovh.

If by chance you are familiar with it,  I've noticed that there are a lot of stuttering in the audio but sometimes it is good. Is this because of it does not sync like shairport-sync does or the implementation of syncing is different? 

"Digging into the code of Shairport 1.0 reveals that Shairport 1.0 doesn't use the extra timing information Apple added to AirTunes. (I think this is because Shairport was written originally before the timing information was added in AirTunes 2.) " According to this statement alone. It is also possible that shairplay by juhovh is not implementing Airtunes 2?

Im asking for your opinion because im thinking of using shairport-sync to make it as my Airtunes implementation in my software. Thanks!

Btw, im developing this because it was assigned to me for my project. Thanks! :)
